flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptivebatch/util/PointwiseVertexInputInfoComputerTest.java

```diff
@@ -0,0 +1,186 @@
+        PointwiseVertexInputInfoComputer computer = createPointwiseVertexInputInfoComputer();
+        List<BlockingInputInfo> inputInfosWithDifferentNumSubpartitions = new ArrayList<>();
+        BlockingInputInfo inputInfo1 = createBlockingInputInfo(3, 3, List.of(1), false);
+        BlockingInputInfo inputInfo2 = createBlockingInputInfo(3, 5, List.of(1), false);
+        inputInfosWithDifferentNumSubpartitions.add(inputInfo1);
+        inputInfosWithDifferentNumSubpartitions.add(inputInfo2);
+        Map<IntermediateDataSetID, JobVertexInputInfo> vertexInputs =
+                computer.compute(inputInfosWithDifferentNumSubpartitions, 3, 10);
+        checkCorrectnessForNonCorrelatedInput(vertexInputs, inputInfo1, 3);
+        checkConsumedDataVolumePerSubtask(
+                new long[] {13L, 10L, 13L}, List.of(inputInfo1), vertexInputs);
+
+        checkCorrectnessForNonCorrelatedInput(vertexInputs, inputInfo2, 3);
+        checkConsumedDataVolumePerSubtask(
+                new long[] {25L, 20L, 15L}, List.of(inputInfo2), vertexInputs);
+    }
+
+    @Test
+    void testComputeInputWithIntraCorrelation() {
+        PointwiseVertexInputInfoComputer computer = createPointwiseVertexInputInfoComputer();
+        List<BlockingInputInfo> inputInfos = createBlockingInputInfos(3, List.of(), true);
+        Map<IntermediateDataSetID, JobVertexInputInfo> vertexInputs =
+                computer.compute(inputInfos, 3, 10);
+        checkCorrectnessForNonCorrelatedInput(vertexInputs, inputInfos.get(0), 3);
+        checkConsumedSubpartitionGroups(
+                List.of(
+                        Map.of(new IndexRange(0, 0), new IndexRange(0, 2)),
+                        Map.of(new IndexRange(1, 1), new IndexRange(0, 2)),
+                        Map.of(new IndexRange(2, 2), new IndexRange(0, 2))),
+                inputInfos,
+                vertexInputs);
+
+        // with different parallelism
+        Map<IntermediateDataSetID, JobVertexInputInfo> vertexInputs2 =
+                computer.compute(inputInfos, 2, 10);
+        checkCorrectnessForNonCorrelatedInput(vertexInputs2, inputInfos.get(0), 2);
+        checkConsumedSubpartitionGroups(
+                List.of(
+                        Map.of(new IndexRange(0, 1), new IndexRange(0, 2)),
+                        Map.of(new IndexRange(2, 2), new IndexRange(0, 2))),
+                inputInfos,
+                vertexInputs2);
+
+        // with bigger parallelism and fallback to
+        // VertexInputInfoComputationUtils.computeVertexInputInfoForPointwise
+        Map<IntermediateDataSetID, JobVertexInputInfo> vertexInputs3 =
+                computer.compute(inputInfos, 4, 10);
+        checkCorrectnessForNonCorrelatedInput(vertexInputs3, inputInfos.get(0), 4);
+        checkConsumedSubpartitionGroups(
+                List.of(
+                        Map.of(new IndexRange(0, 0), new IndexRange(0, 0)),
+                        Map.of(new IndexRange(0, 0), new IndexRange(1, 2)),
```

Comment:
The data of partition 0 should not be split if the input has intra key correlation.

However, the case with a bigger parallelism and intra key correlation seems to be an invalid case?